app/components/Button.js

```diff
@@ -1,87 +1,89 @@
-      titleStyle,
-      buttonStyle,
-      buttonHeight = 54,
-      borderColor,
-      disabled,
-    } = this.props;
-    return (
-      <LinearGradient
-        start={{ x: 0, y: 0 }}
-        end={{ x: 1, y: 0 }}
-        colors={[bgColor, toBgColor]}
-        style={[
-          buttonStyle ? buttonStyle : styles.container,
-          {
-            height: buttonHeight,
-            borderWidth: borderColor ? 2 : 0,
-            borderColor: borderColor,
-          },
-        ]}>
-        <TouchableOpacity
-          style={[
-            buttonStyle ? buttonStyle : styles.container,
-            { height: buttonHeight },
-          ]}
-          onPress={onPress}
-          accessible
-          accessibilityLabel={title}
-          accessibilityRole='button'
-          disabled={disabled}>
-          <Typography
-            style={[
-              titleStyle ? titleStyle : styles.text,
-              {
-                color: buttonColor,
-              },
-            ]}>
-            {title}
-          </Typography>
-        </TouchableOpacity>
-      </LinearGradient>
-    );
-  }
-}
-
-const styles = StyleSheet.create({
-  container: {
-    width: '100%',
-    alignItems: 'center',
-    justifyContent: 'center',
-    paddingVertical: 8,
-    borderRadius: 12,
-  },
-  text: {
-    textAlign: 'center',
-    height: 28,
-    fontSize: 20,
-  },
-});
+/**
+ * @param {{
+ *   label: string;
+ *   secondary?: boolean;
+ *   icon?: string;
+ *   onPress: () => void;
+ *   disabled?: boolean;
+ * }} param0
+ */
+export const Button = ({
+  label,
+  secondary,
+  icon,
+  onPress,
+  disabled,
+  small,
+  ...otherProps
+}) => {
+  return (
+    <Container
+      onPress={onPress}
+      accessible
+      accessibilityLabel={label}
+      accessibilityRole='button'
+      hasIcon={!!icon}
+      secondary={secondary}
+      disabled={disabled}
+      small={small}
+      {...otherProps}>
+      <ThemeProvider theme={invertTextColors}>
+        <Label small={small} secondary={secondary} disabled={disabled}>
+          {label}
+        </Label>
+        {icon ? <Icon xml={icon} /> : null}
+      </ThemeProvider>
+    </Container>
+  );
+};
 
-Button.propTypes = {
-  title: PropTypes.string.isRequired,
-  onPress: PropTypes.func.isRequired,
-  buttonColor: PropTypes.string,
-  bgColor: PropTypes.string,
-  toBgColor: PropTypes.string,
-  titleStyle: PropTypes.object,
-  buttonStyle: PropTypes.object,
-  borderColor: PropTypes.string,
+const invertTextColors = theme => {
+  return {
+    ...theme,
+    textPrimaryOnBackground: theme.onPrimary,
+    textSecondaryOnBackground: theme.primary,
+  };
 };
 
-export default Button;
+const themePrimary = ({ theme, secondary, disabled }) =>
+  disabled ? theme.disabled : secondary ? 'transparent' : theme.primary;
+
+const getBorderColor = ({ theme, secondary }) =>
+  secondary ? theme.primary : 'transparent';
+
+const getJustifyContent = ({ hasIcon }) =>
+  hasIcon ? 'space-between' : 'center';
```

Comment:
```suggestion
  hasIcon ? 'space-around' : 'center';
```

#### `justify-content: space-between`
<img width="389" alt="Screen Shot 2020-05-04 at 9 49 49 PM" src="https://user-images.githubusercontent.com/20157849/81031221-30c60380-8e51-11ea-8f98-ba6c9887f609.png">

#### `justify-content: space-around`
<img width="384" alt="Screen Shot 2020-05-04 at 9 48 44 PM" src="https://user-images.githubusercontent.com/20157849/81031174-083e0980-8e51-11ea-8801-7fee9bad170f.png">

Reply:
Ah, fixed. I removed padding to fix iOS when there was an icon and didn't check everything. :)

vertical padding caused this: 

<img width="487" alt="Screen Shot 2020-05-04 at 8 00 55 PM" src="https://user-images.githubusercontent.com/702990/81031730-19334e80-8e42-11ea-9004-d3bbbe913c2c.png">


padding-horizontal works:

<img width="487" alt="Screen Shot 2020-05-04 at 8 02 12 PM" src="https://user-images.githubusercontent.com/702990/81031755-2e0fe200-8e42-11ea-9f8e-857514aa4b78.png">
<img width="487" alt="Screen Shot 2020-05-04 at 8 02 15 PM" src="https://user-images.githubusercontent.com/702990/81031761-31a36900-8e42-11ea-86c0-4a7fa6b87135.png">